src/controller/AbstractMdnsDiscoveryController.cpp

```diff
@@ -31,6 +31,8 @@ namespace Controller {
 
 void AbstractMdnsDiscoveryController::OnNodeDiscoveryComplete(const chip::Mdns::DiscoveredNodeData & nodeData)
 {
+    chip::Protocols::UserDirectedCommissioning::UserDirectedCommissioningServer::GetInstance().OnCommissionableNodeFound(nodeData);
```

Comment:
This function may be called when either a commissionable node or a commissioner is found. If you are interested specifically in commissionable nodes consider overriding this function in the CHIPDeviceController, adding this UDCServer:OnCommissionableNodeFound call and calling the super AbstractMdnsDiscoveryController::OnNodeDiscoveryComplete from there.